### DISCUSSION TITLE
Showing "today" instead of "14 hours ago"

### DISCUSSION THREAD

**Discussion by latinvm** [xirixiz/homeassistant-afvalwijzer#321] [Q&A]
Thanks for this great plugin!

I was wondering how I can best change the caption from "14 hours ago" to "today" in this overview:
![image](https://github.com/user-attachments/assets/2a70b968-f44f-41d9-8d47-568d1857de78)

It would also be nice if we could change the icon or color for the waste that's being picked up "today".

**Comment by maceddy** [comment]
I have a similar question. I would like to see the (simple) date. So for tomorrow that would be 10-dec. I have tried some things but can not get it to work...

**Reply by snhnic** [reply]
same

**Comment by Mariusthvdb** [comment]
you can use any time format you chose, since the entities are of `device_class: timestamp`

this allows for easy and complex manipulation in jinja, and set it as such in the Frontend, using the [format option](https://www.home-assistant.io/dashboards/entities/#format)

using `relative` in the Frontend does change it to what you show above, so if that is not exactly s you want it, you need to use another card to display it. 

hop over to the community where I have [posted several examples in my cards](https://community.home-assistant.io/t/garbage-pickup-date-mijnafvalwijzer-nl-custom-component/34631/1003?u=mariusthvdb).
be sure to check the [jinja time template options](https://www.home-assistant.io/docs/configuration/templating/#time) in the documentation and use eg custom `template-entity-row` 


as for the color customization, there are many options, all custom, and most used is card_mod.
otoh, if you use the icons/images I suggest in that post linked above, you no longer need to set a color, as these are already colorized

<img width="600" alt="Scherm­afbeelding 2024-12-19 om 10 31 46" src="https://github.com/user-attachments/assets/c2b1f130-023d-4f33-a7c2-8638bf7233a4" />

<img width="583" alt="Scherm­afbeelding 2024-12-19 om 10 32 26" src="https://github.com/user-attachments/assets/550c3942-d724-486d-adb0-86d9ce10e0b7" />

**Comment by snhnic** [comment]
Still don't get it? this is what I have. 

entities:
  - entity: sensor.afvalwijzer_gft
    name: GFT
    image: https://www.zetart.eu/images/GFT.png
    format: date

**Reply by Mariusthvdb** [reply]
guess you didnt check the links I suggested?  please do as they hold the answer to your ask.

all default options in Home Assistant are listed there. If you want another representation, you need to get creative with templates and use a card that can do/display that

you cant just say I want it t show an unsupported 'today' and expect that to happen without creating that yourself.

templating is your answer to that.

**Comment by snhnic** [comment]
I'm reading but don't understand. I got pain i my neck from reading everything. But don't find the examples for my card. Read about entities fomatted and stuff. Feeling dumb now. So I stop asking and going to give up. Just wanted to see a shorter date, but guess it's to hard for tech guys to give me as a noob a proper answer. All I get from here is "do some reading" Well I did and it's not helpfull. But thanks for trying to help.

**Reply by Mariusthvdb** [reply]
let me try again, so please help us help you:
what exactly do you need?

**Comment by snhnic** [comment]
I got this, but I like to see its with dates like 20-12-2024. I read about mushroomcards but I can't get a glance mushroom? I feel realy dumb. I got a condition with my memory and it's real hard for me to get it. And as I said I do read a lot, I'm trying. 

![garbage](https://github.com/user-attachments/assets/69df0913-843a-4a82-9ad0-0ab7aeb11153)

show_name: true
show_icon: true
show_state: true
type: glance
entities:
  - entity: sensor.afvalwijzer_gft
    name: GFT
    image: /local/images/GFT.png
    format: date
  - entity: sensor.afvalwijzer_restafval
    name: Rest
    image: /local/images/Restafval.png
    format: date
  - entity: sensor.afvalwijzer_papier
    image: /local/images/Papier.png
    name: Papier
    format: date
  - entity: sensor.afvalwijzer_plastic
    image: /local/images/PMD.png
    name: Plastic
    format: date
state_color: false
theme: amoled
grid_options:
  rows: auto
columns: 2

**Reply by Mariusthvdb** [reply]
ok, no problem.

first things first, please format you yaml with the <> button in the editor, or add three back-ticks before and after the yaml. That way we can easily spot yaml errors. so like

```
type: glance
entities:

  - entity: sensor.afvalwijzer_gft
    name: GFT
    image: /local/images/GFT.png
    format: date
```
<img width="248" alt="Scherm­afbeelding 2024-12-19 om 12 23 44" src="https://github.com/user-attachments/assets/a0dd5977-4766-44e1-902c-b8c00452f244" />

**Comment by snhnic** [comment]
Finaly I did it.
`
```
show_name: true
show_icon: true
show_state: true
type: glance
entities:
  - entity: sensor.afvalwijzer_gft
    name: GFT
    image: /local/images/GFT.png
    format: date
  - entity: sensor.afvalwijzer_restafval
    name: Rest
    image: /local/images/Restafval.png
    format: date
  - entity: sensor.afvalwijzer_papier
    image: /local/images/Papier.png
    name: Papier
    format: date
  - entity: sensor.afvalwijzer_plastic
    image: /local/images/PMD.png
    name: Plastic
    format: date
state_color: false
theme: amoled
grid_options:
  rows: auto
```
`

**Reply by Mariusthvdb** [reply]
nope....


select the text and then click the <> button 


see this too which explains it a bit more https://community.home-assistant.io/t/how-to-help-us-help-you-or-how-to-ask-a-good-question/114371#oneone-format-it-properly-16

**Reply by Mariusthvdb** [reply]
bingo
ok now for the more important stuff:

Home Assistant has only a limited set of default options to display datetime entities . If those are not to your liking, you have to 'recreate' the state you want.

given the entities made by Afvalwijzer are of `device_class: timestamp` you can manipulate these using jinja templating ( the link I posted earlier).

You can either create a template sensor, or, use a card that can display these templates in the Frontend. In Core Homeassistant, that is only Markdown card.

There are a number of custom cards however, and of those, I would suggest you to use template-entity-row. It's the one I also use in that dashboard.

there is no limit to what you can do in custom cards. but its not always straightforward, and especially for a beginner can be daunting

check this in the dev tools template

```
{{states('sensor.afvalwijzer_papier')}}


{{states('sensor.afvalwijzer_papier')|as_timestamp|timestamp_custom('%d-%m-%y')}}
```

it should do what you want:

<img width="843" alt="Scherm­afbeelding 2024-12-19 om 12 36 08" src="https://github.com/user-attachments/assets/4c92a1ab-aad5-49ab-a218-2981ea3abf5c" />

or use the %Y in that formatter to get it to show the full year:

<img width="839" alt="Scherm­afbeelding 2024-12-19 om 12 42 45" src="https://github.com/user-attachments/assets/a4142ad9-0bbc-446f-b0bc-c2a9768e5157" />

if yes, than you can add that to a template-entity-row as state

you can also go a bit more detailed, and check the attribute `days_until_collection_date`. If that is set to 0, the collection is today. So, in that case you can have it report 'Today', or 'Vandaag' or, whatever you would like.

that is also what I do in my config I posted in the Community

**Comment by snhnic** [comment]
Okay got template-entity-row but next? 

```
show_name: true
show_icon: true
show_state: true
type: glance
entities:
  - type: custom:template-entity-row
    entity: sensor.afvalwijzer_gft
    name: GFT
    image: /local/images/GFT.png
    format: date
theme: amoled
```

**Reply by Mariusthvdb** [reply]
ok, so now you need to enter a config as is documented in that custom card.... you do need to conform to the 'manual' ;-)
try
```
  - type: custom:template-entity-row
    entity: sensor.afvalwijzer_gft
    state: >
      {{states('sensor.afvalwijzer_gft')|as_timestamp|timestamp_custom('%d-%m-%y')}}
    name: GFT
```
or
```
  - type: custom:template-entity-row
    entity: sensor.afvalwijzer_gft
    state: >
      {{states(config.entity)|as_timestamp|timestamp_custom('%d-%m-%y')}}
    name: GFT
```

shows as:

<img width="344" alt="Scherm­afbeelding 2024-12-19 om 14 05 44" src="https://github.com/user-attachments/assets/964d19e4-9bcb-4367-ad10-b70ba0155b68" />

note for that image, I do set a customize in the yaml configuration

```
homeassistant:

  customize:

    sensor.afvalwijzer_plastic:
      entity_picture: /local/afvalwijzer/saver-petfles-blik-drankpak-pmd.png

    sensor.afvalwijzer_gft:
      entity_picture: /local/afvalwijzer/saver-appel-en-blad.png

    sensor.afvalwijzer_papier:
      entity_picture: /local/afvalwijzer/saver-papier.png

    sensor.afvalwijzer_restafval:
      entity_picture: /local/afvalwijzer/saver-rest-kliko.png
```

**Reply by Mariusthvdb** [reply]
but, given this goes way beyond possible 'issues' for the integration, and in fact is support to you for configuration questions in general, I would prefer not to use this space for that, and go to the community (use that topic thread I posted above)

closing here with:
```
  - type: custom:template-entity-row
    entity: sensor.afvalwijzer_gft
    state: |
      {{states(config.entity)|as_timestamp|timestamp_custom('%d-%m-%Y')}}
    secondary: >
      Over {{state_attr(config.entity,'days_until_collection_date')}} dagen
```

and a nice listing in HA card editor if the various available formats

<img width="1011" alt="Scherm­afbeelding 2024-12-19 om 14 20 06" src="https://github.com/user-attachments/assets/0d423c74-2f5a-45c3-934e-2beaa8eac8f6" />

**Comment by snhnic** [comment]
It works but not on the glance card.  Well this will be fine. Thanks will try to get the images as you suggested.

**Reply by Mariusthvdb** [reply]
of course it won't work on the glance card, as that is very limited in [its options](https://www.home-assistant.io/dashboards/glance/#configuration-variables)

you can't just make up configurations.

**Comment by snhnic** [comment]
Thanks for the support. It looks superb now. 

![IMG_2781](https://github.com/user-attachments/assets/26977ffa-48b0-444f-8621-149051c95562)

**Comment by Largus72** [comment]
Okay, back to the original question: Showing "today" instead of "14 hours ago".
Despite this interesting discussion, I unfortunately can't find the answer to this. I have the same problem and would like to have "Today" instead of "11 hours ago".
Yesterday it said "in x hours". I would rather have "Tomorrow" there. After all, the dishes aren't picked up at exactly 12 o'clock at night :).

![Schermafbeelding](https://github.com/user-attachments/assets/a3ecb43b-f026-4167-bade-3a85ab7bab31)

**Reply by Mariusthvdb** [reply]
You can’t make the sensors output that in frontend, all available options are in the documents
(Well, you could, but you’d have to create a template sensor for that yourself)

but, why so much trouble to have Tomorrow listed twice? You have dedicated sensors for those ( vandaag, morgen, overmorgen) and already show them? They were added for exactly that purpose .

just move to format date to get rid of the relative format

**Reply by Largus72** [reply]
Hi Marius,

I already have format date:

```
  - type: custom:auto-entities
    card:
      type: glance
    filter:
      include:
        - entity_id: sensor.afvalwijzer_gft
        - entity_id: sensor.afvalwijzer_pmd
        - entity_id: sensor.afvalwijzer_restafval
          options:
            format: date
    sort:
      attribute: days_until_collection_date
      method: attribute
      numeric: true
      reverse: false
```


And true.... I don't need Tomorrow twice
But what else should be as third icon.

**Reply by Mariusthvdb** [reply]
just let auto-entities take care of it correctly, as you do it, you list then entities yourself....

```
    - type: custom:auto-entities
      card:
        type: glance
      filter:
        exclude:
          - entity_id: sensor.afvalwijzer*next*
          - entity_id: sensor.afvalwijzer_*orgen
          - entity_id: sensor.afvalwijzer_v*
        include:
          - entity_id: sensor.afvalwijzer_*
            options:
              format: date
      sort:
        method: state
```

is all there is to it (do check the names of the entities, I adapted some of them)

I dont use that glance card anymore, as it tripled the info in my card ;-) However, re-enabling shows what it does:

<img width="517" alt="Scherm­afbeelding 2025-01-03 om 14 14 59" src="https://github.com/user-attachments/assets/c657cd0b-5ac2-486b-8980-253ff53472fe" />

the format should give you the date as in my screenshot, and not anything relative at all. which is what happens because you have set that format only on the 3d listed entity..... as is showing in your screenshot too.
